homeassistant/components/github/sensor.py

```diff
@@ -1,293 +1,355 @@
-
-        if self._github_data.clones_response:
-            self._clones = self._github_data.clones_response.data.count
-            self._clones_unique = self._github_data.clones_response.data.uniques
-
-        if self._github_data.views_response:
-            self._views = self._github_data.views_response.data.count
-            self._views_unique = self._github_data.views_response.data.uniques
-
-        self._attr_extra_state_attributes = {
-            ATTR_PATH: self._github_data.repository_path,
-            ATTR_NAME: self.name,
-            ATTR_LATEST_COMMIT_MESSAGE: self._latest_commit_message,
-            ATTR_LATEST_COMMIT_SHA: self._latest_commit_sha,
-            ATTR_LATEST_RELEASE_URL: self._latest_release_url,
-            ATTR_LATEST_OPEN_ISSUE_URL: self._latest_open_issue_url,
-            ATTR_OPEN_ISSUES: self._open_issue_count,
-            ATTR_LATEST_OPEN_PULL_REQUEST_URL: self._latest_open_pr_url,
-            ATTR_OPEN_PULL_REQUESTS: self._pull_request_count,
-            ATTR_STARGAZERS: self._stargazers,
-            ATTR_FORKS: self._forks,
-        }
-        if self._latest_release_tag is not None:
-            self._attr_extra_state_attributes[
-                ATTR_LATEST_RELEASE_TAG
-            ] = self._latest_release_tag
-        if self._clones is not None:
-            self._attr_extra_state_attributes[ATTR_CLONES] = self._clones
-        if self._clones_unique is not None:
-            self._attr_extra_state_attributes[ATTR_CLONES_UNIQUE] = self._clones_unique
-        if self._views is not None:
-            self._attr_extra_state_attributes[ATTR_VIEWS] = self._views
-        if self._views_unique is not None:
-            self._attr_extra_state_attributes[ATTR_VIEWS_UNIQUE] = self._views_unique
-
-
-class GitHubData:
-    """GitHub Data object."""
-
-    def __init__(self, repository, access_token, session, server_url=None):
-        """Set up GitHub."""
-        self._repository = repository
-        self.repository_path = repository[CONF_PATH]
-        self._github = GitHubAPI(
-            token=access_token, session=session, **{"base_url": server_url}
+    def __init__(
+        self,
+        coordinators: DataUpdateCoordinators,
+        repository_information: GitHubRepositoryModel,
+    ) -> None:
+        """Initialize a GitHub sensor entity."""
+        super().__init__(
+            coordinator=coordinators[self._coordinator_key],
+            repository_information=repository_information,
         )
+        self._attr_name = f"{repository_information.full_name} {self._name}"
+        self._attr_unique_id = (
+            f"{repository_information.id}_{self._name.lower().replace(' ', '_')}"
+        )
+
+
+class GitHubSensorLatestReleaseEntity(GitHubSensorLatestBaseEntity):
+    """Defines a GitHub latest release sensor entity."""
+
+    _coordinator_key: CoordinatorKeyType = "release"
+    _name: str = "Latest Release"
+
+    _attr_entity_registry_enabled_default = True
```

Comment:
This is the only sensor that will be enabled by default, should that also be disabled for consistency?
That way you have full control of what entities are enabled pr. repo

Reply:
No, it should be enabled.